concent_api/conductor/views.py

```diff
@@ -53,13 +53,13 @@ def report_upload(_request: HttpRequest, file_path: str) -> HttpResponse:
     ):
         assert file_path in [verification_request.source_package_path, verification_request.result_package_path]
 
-        # If all expected files have been uploaded, the app sends upload_finished task to the work queue.
-        upload_finished.delay(verification_request.subtask_id)
-
         verification_request.upload_finished = True
         verification_request.full_clean()
         verification_request.save()
 
+        # If all expected files have been uploaded, the app sends upload_finished task to the work queue.
+        upload_finished.delay(verification_request.subtask_id)
```

Comment:
What about `update_upload_report()` that's called later? It makes database changes too.

BTW, this is not the subject of this pull request but the structure of this function is really confusing and we should fix it, possibly in a separate pull request:
- `update_upload_report()` does not really modify `UploadReport`. It updates `ResultTransferRequest.upload_finished`
- Why is `ResultTransferRequest` modification hidden in a function while for `VerificationRequest` it's all out in the open? These to cases are symmetric: we either deal with an upload request or with a verification request; in both cases we have an upload report, use it to find and update the request, then fire a Celery task. The code should reflect that. I had to spend a few minutes to understand what's actually happening here.

Reply:
@cameel 
> update_upload_report() does not really modify UploadReport. It updates ResultTransferRequest.upload_finished

it does with this code:

```
UploadReport.objects.select_for_update().filter(
        path=file_path
    ).update(
        result_transfer_request=result_transfer_request
    )
```

> Why is ResultTransferRequest modification hidden in a function while for VerificationRequest it's all out in the open?

They were done in two different issues with some time interval between them, and `update_upload_report` is reused twice in both view and task.

> What about update_upload_report() that's called later? It makes database changes too.

Yes, but the transaction will still exist when those changes from `update_upload_report` are being done, and after they are all done the task will be scheduled.